### DISCUSSION TITLE
Looking for a way to collapse JBrowse categories by default

### DISCUSSION THREAD

**Discussion by jdepons** [GMOD/jbrowse-components#4550] [General] [answered]
Is there a way to tell JBrowse 2 to collapse all the track categories by default?   I have 19 different assemblies that include up to 100 categories each.   The category names are not consistent and differ between assemblies.  I’ve listed a section of my default config for one of the assemblies below.   Is there a way to set “collapsed=true” as a default or do I need to explicitly list every category as listed below?  Currently, if a new category is added or a category name changes I need to update the config.  

Any help is appreciated as it is difficult to mange this for 19 different assemblies.  

"widgets": {
      "hierarchicalTrackSelector": {
        "id": "hierarchicalTrackSelector",
        "type": "HierarchicalTrackSelectorWidget",
        "initialized": true,
        "collapsed": {
          "Disease Related Tracks,Animal Diseases": true,
          "Disease Related Tracks,Bacterial Infections and Mycoses": true,
          "Disease Related Tracks,Cardiovascular Diseases": true,
          "Disease Related Tracks,Congenital-Hereditary-Neonatal Diseases and Abnormalities": true,
          "Disease Related Tracks,Developmental Diseases": true,
          "Disease Related Tracks,Digestive System Diseases": true,
          "Disease Related Tracks,Diseases of the Aged": true,
          "Disease Related Tracks,Endocrine System Diseases": true,
          "Disease Related Tracks,Eye Diseases": true,
          "Disease Related Tracks,Female Urogenital Diseases and Pregnancy Complications": true,
          "Disease Related Tracks,Hemic and Lymphatic Diseases": true,
          "Disease Related Tracks,Immune System Diseases": true,
          "Disease Related Tracks,Induced Disorders": true,
          "Disease Related Tracks,Male Urogenital Diseases": true,
          "Disease Related Tracks,Mental Disorders": true,
          "Disease Related Tracks,Musculoskeletal Diseases": true,
          "Disease Related Tracks,Neoplasms": true,
          "Disease Related Tracks,Nervous System Diseases": true,
          "Disease Related Tracks,Nutritional and Metabolic Diseases": true,
          "Disease Related Tracks,Occupational Diseases": true,
          "Disease Related Tracks,Otorhinolaryngologic Diseases": true,
          "Disease Related Tracks,Parasitic Diseases": true,
          "Disease Related Tracks,Respiratory Tract Diseases": true,
          "Disease Related Tracks,Signs and Symptoms": true,
          "Disease Related Tracks,Skin and Connective Tissue Diseases": true,
          "Disease Related Tracks,Stomatognathic Diseases": true,
          "Disease Related Tracks,Virus Diseases": true,
          "Disease Related Tracks,Wounds and Injuries": true,
          "Dwinell,F344_DuCrl_Gastrocnemius_F": true,
          "Dwinell,F344_DuCrl_Gastrocnemius_M": true,
          "Dwinell,F344_DuCrl_LRetro Fat Pad_F": true,
          "Dwinell,F344_DuCrl_LRetro Fat Pad_M": true,
          "Dwinell,F344_DuCrl_Thymus_F": true,
          "Dwinell,F344_DuCrl_Thymus_M": true,
          "Dwinell,LE_StmMcwi_Gastrocnemius_F": true,
          "Dwinell,LE_StmMcwi_Gastrocnemius_M": true,
          "Dwinell,LE_StmMcwi_LRetro Fat Pad_F": true,
          "Dwinell,LE_StmMcwi_LRetro Fat Pad_M": true,
          "Dwinell,LE_StmMcwi_Thymus_F": true,
          "Dwinell,LE_StmMcwi_Thymus_M": true,
          "Epigenetics,ATAC-Seq,Adrenal_Gland": true,
          "Epigenetics,ATAC-Seq,Amygdala": true,
          "Epigenetics,ATAC-Seq,Auditory_Cortex": true,
          "Epigenetics,ATAC-Seq,Cerebellum": true,
          "Epigenetics,ATAC-Seq,Epididymis": true,
          "Epigenetics,ATAC-Seq,Heart": true,
          "Epigenetics,ATAC-Seq,Hippocampus": true,
          "Epigenetics,ATAC-Seq,Hypothalamus": true,
          "Epigenetics,ATAC-Seq,Ileum": true,
          "Epigenetics,ATAC-Seq,Kidney": true,
          "Epigenetics,ATAC-Seq,Liver": true,
          "Epigenetics,ATAC-Seq,Lung": true,
          "Epigenetics,ATAC-Seq,Motor_Cortex": true,
          "Epigenetics,ATAC-Seq,Olfactory_Bulb": true,
          "Epigenetics,ATAC-Seq,Ovary": true,
          "Epigenetics,ATAC-Seq,Pancreas": true,
          "Epigenetics,ATAC-Seq,Prefrontal_Cortex": true,
          "Epigenetics,ATAC-Seq,Primary_Visual_Cortex": true,
          "Epigenetics,ATAC-Seq,Somatosensory_Cortex": true,
          "Epigenetics,ATAC-Seq,Spermaduct": true,
          "Epigenetics,ATAC-Seq,Spleen": true,
          "Epigenetics,ATAC-Seq,Striatum": true,
          "Epigenetics,ATAC-Seq,Testis": true,
          "Epigenetics,ATAC-Seq,Thalamus": true,
          "Epigenetics,ATAC-Seq,Thymus": true,
          "Epigenetics,ChIP-Seq,Denervated Gastrocnemius Muscle": true,
          "Epigenetics,ChIP-Seq,Gastrocnemius Muscle": true,
          "Epigenetics,ChIP-Seq,Liver": true,
          "Gene Models,Ensembl Gene Features": true,
          "Gene Models,NCBI Gene Features": false,
          "Gene-Chemical Interaction Tracks,Biological Role,Aetiopathogenetic Role": true,
          "Gene-Chemical Interaction Tracks,Biological Role,Antigen": true,
          "Gene-Chemical Interaction Tracks,Biological Role,Antimicrobial Agent": true,
          "Gene-Chemical Interaction Tracks,Biological Role,Antimutagen": true,
          "Gene-Chemical Interaction Tracks,Biological Role,Biochemical Role": true,
          "Gene-Chemical Interaction Tracks,Biological Role,Biophysical Role": true,
          "Gene-Chemical Interaction Tracks,Biological Role,Epitope": true,
          "Gene-Chemical Interaction Tracks,Biological Role,Fixative": true,
          "Gene-Chemical Interaction Tracks,Biological Role,Glutathione Depleting Agent": true,
          "Gene-Chemical Interaction Tracks,Biological Role,Growth Regulator": true,
          "Gene-Chemical Interaction Tracks,Biological Role,Hapten": true,
          "Gene-Chemical Interaction Tracks,Biological Role,Immunomodulator": true,
          "Gene-Chemical Interaction Tracks,Biological Role,Inhibitor": true,
          "Gene-Chemical Interaction Tracks,Biological Role,Insect Attractant": true,
          "Gene-Chemical Interaction Tracks,Biological Role,Mitogen": true,
          "Gene-Chemical Interaction Tracks,Biological Role,Molecular Messenger": true,
          "Gene-Chemical Interaction Tracks,Biological Role,Osmolyte": true,
          "Gene-Chemical Interaction Tracks,Biological Role,Pharmacological Role": true,
          "Gene-Chemical Interaction Tracks,Biological Role,Physiological Role": true,
          "Gene-Chemical Interaction Tracks,Biological Role,Poison": true,
          "Gene-Chemical Interaction Tracks,Biological Role,Prohormone": true,
          "Gene-Chemical Interaction Tracks,Biological Role,Provitamin": true,
          "Gene-Chemical Interaction Tracks,Biological Role,Specialized Proresolving Mediator": true,
          "Gene-Chemical Interaction Tracks,Biological Role,Xenobiotic": true,
          "QTLs": true,
          "RNA-Seq": true,
          "Strains": true,
          "Synteny": true,
          "Variants,Damaging Variants": true,
          "Variants,European Variation Archive": true,
          "Variants,Micro Satellite Markers": true,
          "Variants,Phenotypic Alleles Variants": true,
          "Variants,Strain Specific Variants": true,
          "Tracks": false,
          "Disease Related Tracks": true,
          "Dwinell": true,
          "Epigenetics": true,
          "Gene Models": false,
          "Gene-Chemical Interaction Tracks": true,
          "Variants": true,
          "Tracks-Disease Related Tracks": true,
          "Tracks-Epigenetics": true,
          "Tracks-Gene-Chemical Interaction Tracks,Biological Role": true,
          "Tracks-Gene-Chemical Interaction Tracks": true,
          "Tracks-QTLs": true,
          "Tracks-RNA-Seq": true,
          "Tracks-Gene Models": true,
          "Tracks-Orthologs": true,
          "Tracks-Strains": true,
          "Tracks-Synteny": true,
          "Tracks-Variants": true
        },

**Comment by cmdcolin** [comment, marked as answer]
so it looks like your setting that you have shown is in the "session state" part of the config (e.g. in the defaultSession)

there is instead a configuration slot for this https://jbrowse.org/jb2/docs/config_guides/track_selector/

here is an example of a config.json that collapses all "top level categories" by default
you can see it looks like this
![image](https://github.com/user-attachments/assets/f787f880-d8f1-4b4f-8b83-93972fd52dc7)


```
{
  "configuration": {
    "hierarchical": {
      "defaultCollapsed": {
        "topLevelCategories": true
      }
    }
  },
  "tracks": [ /* your tracks */], // just putting this here to give you an idea of where it goes in the config
  "assemblies": [ /* your assemblies */],
  "defaultSession": { /* your default session here */}
}
```

here is a config that collapses all the "subcategories" by default... e.g. it lets the user get a glimpse of all the subcategories in the tree. you can see it looks like this
![image](https://github.com/user-attachments/assets/fb9bba15-9fec-41d6-8f96-958c63f76be1)

```
{
  "configuration": {
    "hierarchical": {
      "defaultCollapsed": {
        "subCategories": true
      }
    }
  },
  "tracks": [...yourtracks],
  "assemblies": [...yourassemblies]
}
```


note that these config options are sort of "in tension" with the session. the session state (e.g. when you refresh the page) remembers the category collapsed states, so what you see as collapsed is not always taken from the configuration, sometimes session overrides it. hope that helps

**Reply by cmdcolin** [reply]
note that there have been some bug fixes to this as recently as v2.13.0, so if possible use the latest or recent version if possible:)

**Reply by jdepons** [reply]
I have tried this and it didn't seem to work.   I will upgrade my instance to see if that helps.  This is what I currently have in the config. 

"configuration": {
   "hierarchical": {
      "sort": {
        "trackNames": true,
        "categories": true
      },
      "defaultCollapse": {
        "topLevelCategories": true,
        "subCategories": true
      }
    },

**Reply by jdepons** [reply]
Actually,  I bet the setting in the default session are overriding.   Thank you for your help!

**Reply by cmdcolin** [reply]
it also stores certain  the 'collapsed settings' in a localstorage setting ("cookie"), so try in private tab and/or clear browser localstorage also when testing it

 the fact that the localstorage setting can override the config can be a bit confusing...i could do a little brainstorming on this

**Comment by carolyncaron** [comment]
I thought I'd jump into this thread since I'm seeing a similar issue since upgrading from v2.11.0 to v2.15.4. My config settings for setting specific track categories as collapsed by default are no longer working on the newest version. Here is my config:
```
  "configuration": {
    "hierarchical": {
      "sort": {
        "trackNames": false,
        "categories": true
      },
      "defaultCollapsed": {
        "categoryNames": [
          "Tracks-Structural Variation,CDC Greenstar",
          "Tracks-Structural Variation,1048-8R",
          "Tracks-Structural Variation,1254S-1",
          "Tracks-Structural Variation,1294M-23",
          "Tracks-Structural Variation,3114-4",
          "Tracks-Structural Variation,3114-31",
          "Tracks-Structural Variation,3114-47",
          "Tracks-Structural Variation,3114-58",
          "Tracks-Structural Variation,3114-69",
          "Tracks-Structural Variation,3114-78",
          "Tracks-Structural Variation,3339-3",
          "Tracks-Structural Variation,964A-46",
          "Tracks-Structural Variation,Blaze-BC",
          "Tracks-Structural Variation,BM4",
          "Tracks-Structural Variation,CDC KR-1",
          "Tracks-Structural Variation,CDC Lupa",
          "Tracks-Structural Variation,CDC Maxim",
          "Tracks-Structural Variation,CDC Milestone",
          "Tracks-Structural Variation,CDC Robin",
          "Tracks-Structural Variation,CDC Viceroy",
          "Tracks-Structural Variation,Eston",
          "Tracks-Structural Variation,IBC 929",
          "Tracks-Structural Variation,ILL 1704",
          "Tracks-Structural Variation,IG 72595",
          "Tracks-Structural Variation,IG 72643"
        ],
        "topLevelCategories": false,
        "subCategories": false
      }
    },
    "theme": {
```
I've tried opening the session in a private window and all tracks are open by default. @colin do you have any ideas, or has the way to specify which categories to collapse changed in between these versions?

**Reply by cmdcolin** [reply]
I believe you can remove the Tracks- prefix. That may have indeed changed between v2.11.0 and v2.15.4! sorry about that

**Reply by carolyncaron** [reply]
Thanks!! That fixed the problem!
